src/main/java/com/gmail/goosius/siegewar/utils/SiegeWarDominationAwardsUtil.java

```diff
@@ -0,0 +1,266 @@
+                try{
+                    nation = nations.get(nationPosition);            
+                    //Gib money
+                    nation.getAccount().deposit(moneyToGrant.get(nationPosition), "Global Domination Award");    
+                    //Gib artifacts
+                    grantArtefactsToNation(artefactsToGrant.get(nationPosition), nation);
+                    //Global message
+                    System.out.println("Global Domination Awards Granted");
+                } catch(Throwable t) {
+                    SiegeWar.severe("Problem granting global domination award to nation " + nation.getName());
+                    SiegeWar.severe(t.getMessage());
+                    t.printStackTrace();
+                }
+            }
+
+            //Remove all domination records
+            nations = new ArrayList<>(TownyUniverse.getInstance().getNations());
+            for(Nation nationForRecordRemoval: nations) {
+                NationMetaDataController.setDominationRecord(nationForRecordRemoval, Collections.emptyList());
+                nationForRecordRemoval.save();
+            }
+        }       
+    }
+
+    private static List<Nation> cullNationsWithTooFewDominationRecords(List<Nation> nations) {
+        int requiredRecords = SiegeWarSettings.getDominationAwardsGlobalMinimumAssessmentPeriodHours();
+        List<Nation> result = new ArrayList<>();
+        for(Nation nation: nations) {
+            if(NationMetaDataController.getDominationRecord(nation).size() >= requiredRecords)
+                result.add(nation);
+        }
+        return result;
+    }
+
+    private static void grantArtefactsToNation(List<Integer> offersToGrantFromEachTier, Nation nation) {
+        List<ItemStack> artefactsToGrant = new ArrayList<>();
+        for(int tier = 0; tier < offersToGrantFromEachTier.size(); tier++) {
+            artefactsToGrant.addAll(generateArtefacts(tier, offersToGrantFromEachTier.get(tier)));
+        }
+        //Get chunk of capital homeblock
+        WorldCoord homeBlockCoord = nation.getCapital().getHomeBlockOrNull().getWorldCoord();
+        Location homeBlockLocation = new Location(homeBlockCoord.getBukkitWorld(), 
+                                                    homeBlockCoord.getX() * TownySettings.getTownBlockSize(),
+                                                    64,
+                                                    homeBlockCoord.getZ()* TownySettings.getTownBlockSize());
+        Chunk homeBlockChunk = homeBlockLocation.getChunk();
+        //Drop artefacts into chests               
+        SiegeWar.getSiegeWar().getServer().getScheduler().runTask(SiegeWar.getSiegeWar(), ()->  dropItemsInChests(homeBlockChunk, artefactsToGrant));
+    }
+
+    private static void dropItemsInChests(Chunk chunk, List<ItemStack> itemsToGrant) {
+        try {
+            chunk.setForceLoaded(true);
+            chunk.load();
```

Comment:
Towny's WorldCoord object has a loadChunks() method which should be used here as it will use PaperLib's getChunkAtAsync.

Reply:
Done in next push